src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -981,6 +1001,7 @@ private static unsafe uint HandleEventRecv(State state, ref StreamEvent evt)
                 switch (state.ReadState)
                 {
                     case ReadState.None:
+                    case ReadState.PendingReadFinished:
```

Comment:
I assume there's a race that makes it possible to enter the callback and the lock it this state?

Reply:
Yes, the callback simply fires twice in a row before the other thread has time to reset the ReadState. Do you think this needs a comment in the code?

Reply:
Yes, especially with current comment only explaining "None" state 😄

Are you sure the reading flow is fine in that case?

And - just on a side note - oh god I have a bad feeling about that 😢 the states becoming a bit blurred and still not fully helping us serialize stuff in a way we want to. ☹️